File: check_features.js
const sqlite3 = require('sqlite3').verbose();

const db = new sqlite3.Database('features.db');

db.serialize(() => {
    // Get stats
    db.get('SELECT COUNT(*) as count FROM features WHERE passes = 1', (err, row) => {
        if (err) {
            console.error('Error:', err);
            return;
        }
        console.log(`Passing: ${row.count}`);
    });

    db.get('SELECT COUNT(*) as count FROM features', (err, row) => {
        if (err) {
            console.error('Error:', err);
            return;
        }
        console.log(`Total: ${row.count}`);
    });

    db.get('SELECT COUNT(*) as count FROM features WHERE in_progress = 1', (err, row) => {
        if (err) {
            console.error('Error:', err);
            return;
        }
        console.log(`In Progress: ${row.count}`);
    });

    // Get a random passing feature
    db.get(`
        SELECT id, category, name, description, steps
        FROM features
        WHERE passes = 1 AND in_progress = 0
        ORDER BY RANDOM()
        LIMIT 1
    `, (err, row) => {
        if (err) {
            console.error('Error:', err);
            db.close();
            return;
        }

        if (row) {
            console.log('\n--- Random passing feature for testing ---');
            console.log(`ID: ${row.id}`);
            console.log(`Category: ${row.category}`);
            console.log(`Name: ${row.name}`);
            console.log(`Description: ${row.description}`);
            console.log(`Steps: ${row.steps}`);
        } else {
            console.log('\nNo passing features available for testing');
        }

        db.close();
    });
});

File: testing-docs/check_features.py
import sqlite3

conn = sqlite3.connect('features.db')
cursor = conn.cursor()

# Get stats
cursor.execute('SELECT COUNT(*) FROM features WHERE passes = 1')
passing = cursor.fetchone()[0]

cursor.execute('SELECT COUNT(*) FROM features')
total = cursor.fetchone()[0]

cursor.execute('SELECT COUNT(*) FROM features WHERE in_progress = 1')
in_progress = cursor.fetchone()[0]

print(f'Passing: {passing}')
print(f'Total: {total}')
print(f'In Progress: {in_progress}')
print(f'Percentage: {(passing/total)*100:.1f}%')

# Get a random passing feature for regression testing
cursor.execute('''
    SELECT id, category, name, description, steps
    FROM features
    WHERE passes = 1 AND in_progress = 0
    ORDER BY RANDOM()
    LIMIT 1
''')

feature = cursor.fetchone()
if feature:
    print(f'\nRandom passing feature for testing:')
    print(f'ID: {feature[0]}')
    print(f'Category: {feature[1]}')
    print(f'Name: {feature[2]}')
    print(f'Description: {feature[3]}')
    print(f'Steps: {feature[4]}')
else:
    print('\nNo passing features available for testing')

conn.close()
